internal/wasi/wasi_test.go

```diff
@@ -484,7 +512,96 @@ func TestAPI_ClockTimeGet_Errors(t *testing.T) {
 // TODO: TestAPI_FdFilestatSetTimes TestAPI_FdFilestatSetTimes_Errors
 // TODO: TestAPI_FdPread TestAPI_FdPread_Errors
 // TODO: TestAPI_FdPrestatGet TestAPI_FdPrestatGet_Errors
-// TODO: TestAPI_FdPrestatDirName TestAPI_FdPrestatDirName_Errors
+
+func TestAPI_FdPrestatDirName(t *testing.T) {
+	opt := Preopen("test", &MemFS{})
+	store, ctx, fn := instantiateWasmStore(t, FunctionFdPrestatDirName, ImportFdPrestatDirName, "test", opt)
+	fd := uint32(3) // fd 3 will be opened for the "test" directory after 0, 1, and 2, that are stdin/out/err
+
+	path := uint32(1)    // arbitrary offset
+	pathLen := uint32(3) // arbitrary length
```

Comment:
when you make the literal map, please use len(expectedDirName) here instead of arbitrary